assets/js/state/sagas/hosts.js

```diff
@@ -1,7 +1,38 @@
-import { put, takeEvery } from 'redux-saga/effects';
-import { HOST_DEREGISTERED, removeHost } from '@state/hosts';
+import { delay, put, race, call, take, takeEvery } from 'redux-saga/effects';
+import {
+  CHECK_HOST_IS_DEREGISTERABLE,
+  CANCEL_CHECK_HOST_IS_DEREGISTERABLE,
+  HOST_DEREGISTERED,
+  removeHost,
+  setHostsDeregisterable,
+} from '@state/hosts';
 import { notify } from '@state/actions/notifications';
 
+export function* markDeregisterableHosts(hosts) {
+  yield put(
+    setHostsDeregisterable(
+      hosts.filter(({ heartbeat }) => heartbeat !== 'passing')
```

Comment:
I am unsure if this logic should be the responsibility of the Sagas, or the Reducer. Thoughts?